pretty_tables/tables.py

```diff
@@ -1,51 +1,119 @@
-        empty_cell_placeholder = whatever you want in place of an empty cell
+        Args:
+            headers (list): a list of headers for the table.
+            rows (list[list]): A list of rows for the table.
+            empty_cell_placeholder: The value to be placed in cells with no data.
+            colors (list): A list of TableColors for each row.
+            truthy (int): Index of the column to be compared for truthy values. If truthy is not None
+                truth comparison will override all other formats.
         """
-        cls._validate_table_input(headers, rows)
-        table = []
-        table.append(headers)
+        PrettyTables._validate_table_input(headers, rows)
+        table = [headers]
         for row in rows:
             formatted_row = []
             for item in row:
-                if item is None:
-                    item = empty_cell_placeholder
-                formatted_row.append(str(item))
+                if item is True:
+                    data = True
+                elif item is False:
+                    data = False
+                elif item:
+                    data = item
+                else:
+                    data = empty_cell_placeholder
+                formatted_row.append(data)
             table.append(formatted_row)
-        formatted_table = cls._format_table(table)
+        formatted_table = cls._format_table(table, colors, truthy)
         return formatted_table
 
-    @classmethod
-    def _format_table(cls, table):
-        """Take table data and format it into a pretty table
-        including lines for columns and uniform spacing.
+    @staticmethod
+    def _format_table(table, colors=None, truthy=None):
+        """Take table data and format it into a pretty table.
+
+        This includes adding lines for columns and uniform spacing.
+
+        Args:
+            table: The list defining the table to be formatted.
+            colors (list): A list of TableColors for each row. If truthy is set to True, colors should
+                be set to None, or a list of colors length 2, where the first color is to be used for
+                true values and the second color is to be used for false values.
+            truthy (int): Index of the column to be compared for truthy values. If truthy is not None
+                truth comparison will override all other formats.
         """
         complete_table = []
         col_widths = [max(len(str(item)) for item in column) for column in zip(*table)]
-        for line in table:
-            if line == table[1]:
-                # Add horizontal separator between headers and rows
-                complete_table.append('| ' + ' | '.join('{:{}}'.format('-' * col_widths[i], col_widths[i])
+        for line_no, line in enumerate(table):
+            if colors or truthy is not None:
+                if truthy is not None:
+                    if colors is not None and (len(colors) != 0 and len(colors) != 2):
```

Comment:
Is there a better way we can lay this out instead of the 3 nested if statements that seem to be doing similar things? It makes it really difficult to read through the code. If there's not a better way, maybe a simple comment to describe the breakdown of if/else logic here to assist?

Could we also just assert a truth value? `if colors or truthy`, or do we need to specifically be checking for `not None` here?